wallet-api/src/Wallet/Emulator/Types.hs

```diff
@@ -383,29 +389,43 @@ evalEmulated = \case
     WalletRecvNotification wallet trigger -> fst <$> liftMockWallet wallet (handleNotifications trigger)
     BlockchainProcessPending -> do
         emState <- get
-        let (block, events) = validateBlock emState (_txPool emState)
+        let (ValidatedBlock block events rest) = validateBlock emState (_txPool emState)
             newChain = block : _chainNewestFirst emState
         put emState {
             _chainNewestFirst = newChain,
-            _txPool = [],
+            _txPool = rest,
             _index = Index.insertBlock block (_index emState),
             _emulatorLog   = SlotAdd (lastSlot newChain) : events ++ _emulatorLog emState
             }
         pure block
     Assertion a -> assert a
 
+-- | Result of validating a block
+data ValidatedBlock = ValidatedBlock
+    { vlbValid  :: [Tx]
+    -- ^ Transactions that have been validated
+    , vlbEvents :: [EmulatorEvent]
+    -- ^ Transaction validation events
+    , vlbRest   :: [Tx]
```

Comment:
Why shouldn't these just be treated like other failed transactions?

Reply:
Previously we validated all transactions in the pool of pending transactions, but now we select those that can be validated in the current slot, and put the rest in `vlbRest` so that they go back into the pool (l. 396)

Reply:
Oh I see, these are ones which can't be validated *yet*. We should probably put ones whose interval has passed completely into the failed transactions, though.